### PR TITLE
revamp of the RPC using the new bsonrpc package

### PR DESCRIPTION
**Pull request by skelterjohn**
Added a bsonrpc package and reworked the service and admin RPCs to use it. Service RPC also wraps a request to provide a special RequestInfo, and forwards it to a method with the right name and signature.

### CODE REVIEW

**Comment by skelterjohn** [comment]
Oh, this also merges in the mux() methods used by a service to safely manage concurrent RPC and doozer requests.

File: skylib/doozer.go [hunk 1]
```diff
@@ -2,7 +2,7 @@ package skylib
 
 import (
 	"bytes"
-	"github.com/4ad/doozer"
+	"github.com/skelterjohn/doozer"
```
Comment (bketelsen) [inline skylib/doozer.go]: what are your changes to doozer client?  Anything substantial?

Comment (skelterjohn) [inline skylib/doozer.go]: I did some stuff with recompiling the protobufs, but 4ad has merged that in. I'll switch back to his import.

**Comment by skelterjohn** [comment]
PTAL

File: skylib/doozer.go (hunk 1 above)
Comment (skelterjohn) [inline skylib/doozer.go]: I do have a more substantial change to doozerd, that supposedly allows it to bring in a new server days after the doozerd cluster is started. Erik gave me the patch - I've got a pull-request in to 4ad/doozerd, but he's sitting on it for the moment (he probably forgot about it).

**Comment by bketelsen** [comment]
LGTM

**Pull request merged**